tests/unit/config/test_loader_edge_cases.py

```
@@ -192,6 +194,142 @@ def test_file_path_included_in_error_context(self) -> None:
+    def test_self_referential_mapping_raises_configuration_error(self) -> None:
+        """A cycle must not escape as a bare RecursionError.
+
+        _assert_string_keys recurses with no cycle guard and runs before
+        expansion, so without the up-front _detect_cycles_or_depth call an
+        in-memory cycle surfaced as RecursionError. Only reachable through this
+        entrypoint: a recursive YAML anchor is caught during expansion instead.
+        """
+        data: dict = {"benchmark": {}}
+        data["benchmark"]["self"] = data
+
+        with pytest.raises(ConfigurationError, match="[Cc]yclic"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)